### PR TITLE
support python 3.12

### PR DESCRIPTION
**Pull request by irexyc**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: src/turbomind/kernels/stop_criteria_kernels.cu [hunk 1]
```diff
@@ -151,11 +151,7 @@ void invokeLengthCriterion(bool*           finished,
 
     length_criterion<<<grid, block, 0, stream>>>(
         finished, should_stop, h_pinned_finished_sum_, sequence_limit_length, batch_size, beam_width, step);
-#ifdef _MSC_VER
```
Comment (lvhan028) [inline src/turbomind/kernels/stop_criteria_kernels.cu:154]: 印象中好像还有其他地方

File: builder/manywheel/Dockerfile_2014 [hunk 2]
```diff
@@ -33,7 +33,9 @@ RUN /opt/conda/bin/conda create -n py38 python=3.8 -yq && \
     /opt/conda/bin/conda create -n py310 python=3.10 -yq && \
     /opt/conda/envs/py310/bin/pip install pybind11 && \
     /opt/conda/bin/conda create -n py311 python=3.11 -yq && \
-    /opt/conda/envs/py311/bin/pip install pybind11
+    /opt/conda/envs/py311/bin/pip install pybind11 && \
```
Comment (lvhan028) [inline builder/manywheel/Dockerfile_2014:36]: docker hub上的镜像有更新么？

File: src/turbomind/kernels/stop_criteria_kernels.cu (hunk 1 above)
Comment (irexyc) [reply src/turbomind/kernels/stop_criteria_kernels.cu:154]: 有两处，另外一个地方不会调用到，之前也删掉了。

File: builder/manywheel/Dockerfile_2014 (hunk 2 above)
Comment (irexyc) [reply builder/manywheel/Dockerfile_2014:36]: 更新过了

**Review by lvhan028 — APPROVED** [review]

**Pull request merged**